Add header and source of OTA MQTT component

pvyawaha opened the pull request:
*Issue #, if available:*

*Description of changes:* The change adds the internal OTA MQTT components , the header and implementation. This component handles the control and well as data over MQTT for Over-the-Air updates.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

pvyawaha committed:
Add header and source of OTA MQTT component

pvyawaha committed:
Fix spell check

pvyawaha committed:
Remove OTA method name

pvyawaha committed:
Remove comment to fix spell check

divekarshubham committed:
Doxygen fix

divekarshubham committed:
Doxygen fix

divekarshubham committed:
spell check fix

divekarshubham committed:
Added missing keywords for spell check

divekarshubham reviewed (dismissed):
Here are some common comments:
1. Should check that msgBuffer is not null in all functions
2. Should change the implementation and the use of snprintf
3. Remove dynamic memory allocations

yanjos-dev committed:
Update ota_mqtt.c log strings

pvyawaha committed:
Change interface prefix

yanjos-dev committed:
Remove unused constants from ota_mqtt.c

yanjos-dev committed:
Update subscribeToJobNotificationTopics and logs

yanjos-dev committed:
Update subscribeToJobNotificationTopics

yanjos-dev committed:
Remove unused parameters from publishStatusMessage

yanjos-dev committed:
Fix incorrect variable name in publishStatusMessage

yanjos-dev committed:
Add asserts for ota_mqtt.c functions

yanjos-dev committed:
Add a preprocessor check for the topic buffer length

yanjos-dev committed:
Update error handling in unsubscribeFromDataStream

yanjos-dev committed:
Update error handling in ota_mqtt.c

For the following functions:
* Update unsubscribeFromDataStream
* Update unsubscribeFromJobNotificationTopic
* Update publishStatusMessage
* Update buildStatusMessageReceiving
* Update prvBuildStatusMessageSelfTest
* Update prvBuildStatusMessageFinish
* Update requestJob_Mqtt
* Update updateJobStatus_Mqtt
* Update initFileTransfer_Mqtt
* Update requestFileBlock_Mqtt
* Update decodeFileBlock_Mqtt
* Update cleanupControl_Mqtt
* Update cleanupData_Mqtt

Make the following changes:
* Use asserts to check for conditions that are assumed to never happen.
* Lower the GNU complexity score to be equal to or less than 8.
* Make it so that the first error that occurs is the one that's reported
  and that the following steps are not executed.
* Have a single variable for tracking error and return values.
* Remove redundant checks.
* Remove checks for the file handle since it's always valid when these
  functions are called.
* Use asserts when using snprintf since the size of the parameters are
  known at compile time.
* Move CBOR error handling into the CBOR library.

yanjos-dev committed:
Fix missing parenthesis in ota_mqtt.c

yanjos-dev committed:
Run uncrustify v0.67 on ota_mqtt.c

yanjos-dev committed:
Empty commit to run CI

yanjos-dev committed:
Fix spelling errors in ota_mqtt.c

yanjos-dev committed:
Add type specifiers to constants in ota_mqtt.c

yanjos-dev committed:
Remove unnecessary asserts

yanjos-dev committed:
Update memory checks in MQTT block decode

yanjos-dev committed:
Fix doxygen warnings in ota_mqtt.c

pvyawaha committed:
Add OTA MQTT interface file

yanjos-dev committed:
Add comments to OtaMqttInterface_t

yanjos-dev committed:
Fix spelling and formatting for ota_mqtt_interface.h

yanjos-dev committed:
Update snprintf related buffers and asserts

* Change the buffer sizes to be specific to the topic being
  built. This allows the buffers to scale with the various parameters
  that determine each topics maximum size.
* Remove the preprocessor check limiting the maximum size of the thing
  name.
* Update snprintf asserts to check for buffers being too small.

yanjos-dev committed:
Fix spelling errors in ota_mqtt.c

Commit:
Merge branch 'main' into ota_mqtt_component

yanjos-dev committed:
Merge branch 'main' into ota_mqtt_component
